### PR TITLE
Pjf 28 dockerize controller

### PR DESCRIPTION
**Pull request by vitorestevamia**

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -42,22 +56,22 @@ services:
...
 
-  temp_collector:
-    container_name: temp_collector
-    image: elazzar/rpi-dht11-api-docker
-    privileged: true
-    ports: 
-      - 5000:5000
-    profiles:
-      - raspberry
+  # temp_collector:
```
Comment (elazzarin) [inline docker/docker-compose.yml:70]: I think it's better to leave the code uncommented, so the real data from the sensors embedded in the raspberry will be collected by the container created

**Review by elazzarin — APPROVED** [review]

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -42,22 +56,22 @@ services:
     depends_on: #required while image doesnt support arm architecture
```
Comment (AnselmoLacerda) [inline docker/docker-compose.yml:56]:
```suggestion
    depends_on: #required while the image doesn't support arm architecture
```

File: docker/dockerfile.controller [hunk 3]
```diff
@@ -0,0 +1,28 @@
+FROM golang:1.20-alpine AS build_base
+
+# Set envs
+# ENV SERVER-AGENT-IP 192.168.0.12
+
+RUN apk add --no-cache git
+
+WORKDIR /tmp/janus
+
+#Copy whole repository
```
Comment (AnselmoLacerda) [inline docker/dockerfile.controller:10]:
```suggestion
#Copy the whole repository
```

File: pkg/agent_deploy/agent.go [hunk 4]
```diff
@@ -8,48 +8,48 @@ import (
...
+
+	//Janus-controller
+	cmd.Env = append(cmd.Env, fmt.Sprintf("CONTROLLER_PORT=%s", info.ControllerPort))
 }
 
-func InstantiateAgent(agent AgentInfo, hostName, profile string, allowRedeploy bool) error {
+func InstantiateAgent(agent ComposeInfo, hostName, profile string) error {
 	command := "docker "
 
 	// add -H host name if remote deploying
```
Comment (AnselmoLacerda) [inline pkg/agent_deploy/agent.go:38]:
```suggestion
	// add -H hostname if remote deploying
```

**Review by AnselmoLacerda — APPROVED** [review]

**Pull request merged**